### DISCUSSION TITLE
Can Odin compiler create libraries that c or c++ can load (.dll, .lib, .a, .so)?

### DISCUSSION THREAD

**Discussion by bovacu** [odin-lang/Odin#2508] [Q&A]
I'm completely new to the language and this might be a silly question or could have been responded on other discussion, but I could not find that information.

That is basically the question, can the odin compiler generate a library that then can be used with C or C++? I know it can be done the other way around, create a C/C++ code base, generate a .lib and then load it onto Odin via foreign blocks and imports, but can Odin generate a dll (from Odin code) that can be loaded into C/C++ code? If not, is there a plan on this?

**Comment by gingerBill** [comment]
Yes. `-build-mode:<string>` is the command you are looking for.
